source/server/listener_impl.cc

```diff
@@ -536,5 +628,80 @@ void ListenerImpl::setSocketFactory(const Network::ListenSocketFactorySharedPtr&
   socket_factory_ = socket_factory;
 }
 
+bool ListenerImpl::supportUpdateFilterChain(const envoy::config::listener::v3::Listener& config,
+                                            bool worker_started) {
+  // If a listener is in active list but stopped, the in place update will fail. The only known case
+  // is that listener is stopped by admin interface. It is ok if that admin method is called only
+  // prior to stop envoy. Otherwise we need to track the stopping state and execute full listener
+  // update.
```

Comment:
It seems bad that this will fail in a way that may not be clear. Can this be a TODO? Should there be some check that we don't update stropped listeners at all? Some ASSERT to warn us? Etc.?

Reply:
I can add a TODO. Basically we need a flag to track the listener in stopped state and populate in the stop path.

Reply:
I personally feel the logic is far away from the MVP, especially under the condition that the PR is growing bigger...
Now that you require, I can add the warning in this PR